File: kotlin/gradle-conventions/src/main/kotlin/software/onepiece/gradle/pluginshowto/MyJavaApplicationPlugin.kt
package software.onepiece.gradle.pluginshowto

import org.gradle.api.Plugin
import org.gradle.api.Project
import org.gradle.api.plugins.ApplicationPlugin

abstract class MyJavaApplicationPlugin : Plugin<Project> {

    override fun apply(project: Project): Unit = with(project) {
        plugins.apply(MyJavaBasePlugin::class.java)
        plugins.apply(ApplicationPlugin::class.java)
    }
}

File: scala/gradle-conventions/src/main/scala/software/onepiece/gradle/pluginshowto/MyJavaApplicationPlugin.scala
package software.onepiece.gradle.pluginshowto

import org.gradle.api.{Plugin, Project}
import org.gradle.api.plugins.ApplicationPlugin

abstract class MyJavaApplicationPlugin extends Plugin[Project] {

    override def apply(project: Project): Unit = {
        project.getPlugins.apply(classOf[MyJavaBasePlugin])
        project.getPlugins.apply(classOf[ApplicationPlugin])
    }
}
